src/tbp/monty/frameworks/models/object_model.py

```diff
@@ -534,7 +535,13 @@ def _initialize_and_fill_grid(
         # The offset is set such that the first observed location starts at the
         # center of the grid. To preserve the relative locations, the offset is
         # applied to all following locations.
-        self._initialize_location_mapping(start_location=locations[0])
+        # Sort out nan locations. For some reason the first location is sometimes nan.
+        # TODO - C: Why is there a nan location?
```

Comment:
😄 Well, why is there a NaN location? Is this something we can easily fix now, or is it better to wait?

Reply:
I am not sure :D I thought I had figured it out but then it resurfaced I think.I think it has something to do with the way the buffer fills in nan values when a channel appears at a later step. I can take a closer look again